### PR TITLE
[Pipelining] Add debug logs for batch p2p ops

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #126582
* __->__ #126539

logs from torchtitan:

image

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @yf225 @chauhang @d4l3k

### CODE REVIEW

File: torch/distributed/pipelining/PipelineSchedule.py [hunk 1]
```diff
@@ -195,7 +195,15 @@ def _merge_outputs(self, output_chunks: List[Any]) -> Any:
         )
 
 
-def sorted_batch_isend_irecv(p2p_ops: List[dist.P2POp]) -> Dict[int, dist.Work]:
+def batch_p2p(p2p_ops: List[dist.P2POp], desc: Optional[str] = None):
+    desc_str = f"{desc}, " if desc else ""
+    logger.debug(f"batch_p2p {desc_str}{p2p_ops}")  # noqa: G004
```
Comment (kwen2501) [inline torch/distributed/pipelining/PipelineSchedule.py:200]: Would print of `p2p_ops` cause a tensor sync like the case of printing loss? I don't know. Just asking wildly here.

**Review by kwen2501 — APPROVED** [review]

File: torch/distributed/pipelining/PipelineSchedule.py (hunk 1 above)
Comment (wconstab) [reply torch/distributed/pipelining/PipelineSchedule.py:200]: no, it only prints shape/dtype. no sync.  unless i'm missing something?  (see previous PR)

Comment (kwen2501) [reply torch/distributed/pipelining/PipelineSchedule.py:200]: oh oh, I missed that this PR is on a stack. LGTM

**Review by H-Huang — APPROVED** [review]
LGTM

**Review by kwen2501 — APPROVED** [review]
lgtm

**Pull request closed without merging**